### PR TITLE
Cleaning up Passes

### PR DESCRIPTION
**Pull request by Ravneetsb**
Extracted the simplify function from all the passes into BasePass trait in the base_pass.rs file

Added 2 parameters to the simplify method :-
- timeout_multiplier -> i64 that allows user to define an integer multiplier for the timeout value.
- has_node_limit -> boolean used to create a runner with/without a node_limit and iter_limit of 100,000

Changed large numbers like 100000 to 100_000 to improve readability.

### CODE REVIEW

File: .idea/algebraic_simplification_cesar.iml
Comment (aditink) [inline .idea/algebraic_simplification_cesar.iml:1]: Maybe .idea should be added to the gitignore?

File: .z3-trace
Comment (aditink) [inline .z3-trace:1]: .z3-trace should also be added to the gitignore and not committed.

File: src/cesar/simplify.rs [hunk 1]
```diff
@@ -10,16 +10,17 @@ use crate::cesar::pass6::Pass6;
 use crate::cesar::pass7::Pass7;
 use crate::cesar::pass8::Pass8;
 use crate::cesar::pass9::Pass9;
+use crate::cesar::base_pass::BasePass;
```
Comment (aditink) [inline src/cesar/simplify.rs:13]: Doublecheck if this is required. I don't see why it should be but have not tried removing it and checking it the code still compiles. Ignore this comment if you have already checked that this is required.

**Review by aditink — CHANGES_REQUESTED** [review]
Just a few minor things, looks good otherwise!

File: .z3-trace
Comment (aditink) [reply .z3-trace:1]: .z3-trace inside folders should similarly not be committed.

File: src/cesar/base_pass.rs [hunk 2]
```diff
@@ -0,0 +1,45 @@
+use crate::cesar::{language::PropLang};
+use crate::cesar::config;
+use egg::*;
+
+pub static mut ASSUMPTIONS: String = String::new();
+
```
Comment (aditink) [inline src/cesar/base_pass.rs:6]: Might be a good idea to add comments explaining the purpose of this trait and how to use it.

File: src/cesar/simplify.rs (hunk 1 above)
Comment (Ravneetsb) [reply src/cesar/simplify.rs:13]: Already checked. It is required.

File: src/.z3-trace
Comment (aditink) [inline src/.z3-trace:1]: This file should be removed/added to gitignore

File: src/cesar/.base_pass.rs.swp
Comment (aditink) [inline src/cesar/.base_pass.rs.swp:1]: This file should be removed/added to gitignore

File: src/cesar/.z3-trace
Comment (aditink) [inline src/cesar/.z3-trace:1]: This file should be removed/added to gitignore

**Review by aditink — APPROVED** [review]
Remove some files, lgtm otherwise

File: src/cesar/base_pass.rs (hunk 2 above)
Comment (Ravneetsb) [reply src/cesar/base_pass.rs:6]: Done.

**Pull request merged**